cores/esp8266/core_esp8266_waveform.cpp

```diff
@@ -259,24 +527,58 @@ static ICACHE_RAM_ATTR void timer1Interrupt() {
         // Check for toggles
         int32_t cyclesToGo = wave->nextServiceCycle - now;
         if (cyclesToGo < 0) {
-          waveformState ^= mask;
-          if (waveformState & mask) {
+          uint32_t nextEdgeCycles;
+          uint32_t desired = 0;
+          uint32_t *timeToUpdate;
+          wvfState.waveformState ^= mask;
+          if (wvfState.waveformState & mask) {
             if (i == 16) {
-              GP16O |= 1; // GPIO16 write slow as it's RMW
+              GP16O = 1; // GPIO16 write slow as it's RMW
             } else {
               SetGPIO(mask);
             }
-            wave->nextServiceCycle = now + wave->nextTimeHighCycles;
-            nextEventCycles = min_u32(nextEventCycles, wave->nextTimeHighCycles);
+#ifdef ENABLE_FEEDBACK
+            if (wave->lastEdge) {
+              desired = wave->desiredLowCycles;
+              timeToUpdate = &wave->timeLowCycles;
+            }
+#endif
+            nextEdgeCycles = wave->timeHighCycles;
           } else {
             if (i == 16) {
-              GP16O &= ~1; // GPIO16 write slow as it's RMW
+              GP16O = 0; // GPIO16 write slow as it's RMW
```

Comment:
No RMW now, remove the confusing comment, too.

Reply:
@earlephilhower The GPIO 16 macros in `esp8266_peri.h` are:
```
//GPIO 16 Control Registers
#define GP16O  ESP8266_REG(0x768)
#define GP16E  ESP8266_REG(0x774)
#define GP16I  ESP8266_REG(0x78C)

//GPIO 16 PIN Control Register
#define GP16C  ESP8266_REG(0x790)
#define GPC16  GP16C

//GPIO 16 PIN Function Register
#define GP16F  ESP8266_REG(0x7A0)
#define GPF16  GP16F

//GPIO 16 PIN Function Bits
#define GP16FFS0 0 //Function Select bit 0
#define GP16FFS1 1 //Function Select bit 1
#define GP16FPD  3 //Pulldown
#define GP16FSPD 5 //Sleep Pulldown
#define GP16FFS2 6 //Function Select bit 2
#define GP16FFS(f) (((f) & 0x03) | (((f) & 0x04) << 4))
```
Would you feel confident enough to apply the change to `core_esp8266_wiring_digital.cpp` `__digitalWrite()` as well? Otherwise, I'm not buying it just yet myself in #7022.

Reply:
`digitalWrite` change would be for another PR (but digitalWrite's not really timing critical like this code that's firing 80,000 per second).

I'm pretty confident that only 1 bit is implemented at that address.  Have you ever read anything but 0 for the upper bits in GP16O that makes you concerned?